opentracing-observer/src/main/java/io/opentracing/contrib/observer/SpanObserver.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.observer;
+
+import java.util.Map;
+
+/**
+ * This interface represents an observer used to receive notifications related to a {@link io.opentracing.Span}.
+ *
+ */
+public interface SpanObserver {
+
+    /**
+     * Notifies the observer that the operation name has been changed.
+     *
+     * @param spanData The data for the span
+     * @param operationName The new operation name
+     */
+    void onSetOperationName(SpanData spanData, String operationName);
+
+    /**
+     * Notifies the observer that the tag with the supplied key has been set or updated
+     * on a {@link io.opentracing.Span}.
+     *
+     * @param spanData The data for the span
+     * @param key The tag key
+     * @param value The tag value
+     */
+    void onSetTag(SpanData spanData, String key, Object value);
```

Comment:
Agree this should be defined - my initial thought is that it is best post change, so that these callbacks could be performed asynchronously to not block the app or tracer. If the observer is interested in the previous state then it could cache the previous values itself.